src/backend/utils/adt/jsonb.c

```diff
@@ -598,6 +598,16 @@ JsonbToCStringWorker(StringInfo out, JsonbContainer *in, int estimated_len, bool
 	return out->data;
 }
 
+void
+jsonb_get_value(Datum val, bool is_null, JsonbValue *json, Oid val_type)
+{
+	JsonbInState state;
+	memset(&state, 0, sizeof(JsonbInState));
+
+	add_jsonb(val, is_null, &state, val_type, false);
+	*json = *(state.res);
```

Comment:
It seems like `json` is palloc'd in `tsql_row_to_json()`. To prevent the possibility of memory overwrites, could you double check whether `json` and `state.res` have the same memory context?

Reply:
The `jsonb_get_value` function is called by `tsql_row_to_json` which defines the `json` variable. The memory context is consistent in `tsql_row_to_json`, so `json` and `state.res`  -  these GDB results were the same after `json` is declared and after `state` is declared 

```
(gdb) print MemoryContextStats(TopMemoryContext)
$5 = void
(gdb) print MemoryContextStats(CacheMemoryContext)
$6 = void
(gdb) print MemoryContextStats(CacheMemoryContext)
$7 = void
(gdb) print MemoryContextStats(PortalContext)
$8 = void
```